From: =?utf-8?q?Florian_M=C3=BCllner?= <fmuellner@gnome.org>
Date: Thu, 17 Sep 2026 16:24:10 +0200
Subject: messageList: Actually limit body/title length

The limit introduced in commit 336a4b3c was only applied to the
property value, but not to the string used as label text. It's
the latter that is more likely to impact performance and should
be limited.

Fixes: 336a4b3c58 ("messageList: Limit body/title length")
Part-of: <https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/4433>

Origin: https://gitlab.gnome.org/GNOME/gnome-shell/-/commit/f8a5dc6
---
 js/ui/messageList.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/js/ui/messageList.js b/js/ui/messageList.js
index 83664f4..f952e39 100644
--- a/js/ui/messageList.js
+++ b/js/ui/messageList.js
@@ -591,7 +591,8 @@ export class Message extends St.Button {
     }
 
     set title(text) {
-        this._titleText = this._limitString(text, MAX_TITLE_LENGTH);
+        text = text ? this._limitString(text, MAX_TITLE_LENGTH) : null;
+        this._titleText = text;
         const title = text ? Util.fixMarkup(text.replace(/\n/g, ' '), false) : '';
         this.titleLabel.clutter_text.set_markup(title);
         this.notify('title');
@@ -602,7 +603,8 @@ export class Message extends St.Button {
     }
 
     set body(text) {
-        this._bodyText = this._limitString(text, MAX_BODY_LENGTH);
+        text = text ? this._limitString(text, MAX_BODY_LENGTH) : null;
+        this._bodyText = text;
         this._bodyLabel.setMarkup(text ? text.replace(/\n/g, ' ') : '',
             this._useBodyMarkup);
         this.notify('body');
